kong/init.lua

```diff
@@ -182,15 +182,15 @@ local function build_new_plugins(version)
   for _, plugin in ipairs(loaded_plugins) do
     if new_plugins.combos[plugin.name] then
       for phase_name, phase in pairs(new_plugins.phases) do
-        if plugin.handler[phase_name] ~= BasePlugin[phase_name] then
+        if type(plugin.handler[phase_name]) == "function"
```

Comment:
just to make sure I follow: why was this change necessary, does any plugin declare a non-function field with a name clashing with a phase handler?

Reply:
The check could have been `if plugin.handler[phase_name] ~= nil`, but this is safer (most likely, plugins handlers should be subject to schema validation, just like migrations are, for instance).

If a handler is not implemented, the `plugin.handler[phase_name] ~= BasePlugin[phase_name]` condition would have been valid, and the phase incorrectly enabled by our runloop.

Reply:
ahhh got it